src/App.tsx

```diff
@@ -1,24 +1,267 @@
+    } catch (error) {
+      setVisibleError(NewError.Remove);
+    } finally {
+      setIsRemovingCompleted(false);
+    }
+  }, [todos]);
+
+  const handleUpdateTodo = useCallback(async (todoToUptade: Todo) => {
+    setUpdatingTodoId(todoToUptade.id);
+    const updatedTodo = {
+      ...todoToUptade,
+      completed: !todoToUptade.completed,
+    };
+
+    try {
+      await updateTodo(updatedTodo);
+
+      setTodos((prevTodos) => prevTodos.map((todo) => (
+        todo.id === todoToUptade.id ? updatedTodo : todo
+      )));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setUpdatingTodoId(null);
+    }
+  }, [todos]);
+
+  const handleUpdateAllTodos = useCallback(async () => {
+    setIsUpdatingEveryStatus(true);
+    try {
+      await Promise.all(
+        todos.map((todo) => updateTodo({
+          ...todo,
+          completed: !isEveryTotoCompleted,
+        })),
+      );
+
+      setTodos((prevTodos) => prevTodos.map((todo) => ({
+        ...todo,
+        completed: !isEveryTotoCompleted,
+      })));
+    } catch (error) {
+      setVisibleError(NewError.Update);
+    } finally {
+      setIsUpdatingEveryStatus(false);
+    }
+  }, [todos]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {isTodosNoEmpty && (
+            <button
+              type="button"
+              className={classNames(
+                'todoapp__toggle-all', {
+                  active: isEveryTotoCompleted,
+                },
+              )}
+              onClick={handleUpdateAllTodos}
+            />
+          )}
+
+          <form onSubmit={handleSubmit}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={query}
+              onChange={(event) => handleQueryChange(event.target.value)}
+            />
+          </form>
+        </header>
+
+        <TodoList
+          todos={visibleTodos}
+          updatingTodoId={updatingTodoId}
+          isRemovingCompleted={isRemovingCompleted}
+          isUpdatingEveryStatus={isUpdatingEveryStatus}
+          isEveryTotoCompleted={isEveryTotoCompleted}
+          onTodoRemove={handleRemoveTodo}
+          onTodoUpdate={handleUpdateTodo}
+        />
+
+        {tempTodo && (
+          <TodoItem
+            todo={tempTodo}
+            isLoadingNewTodo
+            onTodoRemove={() => {}}
+            onTodoUpdate={() => {}}
+          />
```

Comment:
Make a separate `TempTodo` item for it
```suggestion
            todo={tempTodo}
            isLoadingNewTodo
          />
```